src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -838,13 +841,17 @@ ReadFromMergeTree::AnalysisResult ReadFromMergeTree::selectRangesToRead(MergeTre
 
     size_t sum_marks = 0;
     size_t sum_ranges = 0;
+    size_t sum_rows = 0;
 
     for (const auto & part : result.parts_with_ranges)
     {
         sum_ranges += part.ranges.size();
         sum_marks += part.getMarksCount();
+        sum_rows += part.getRowsCount();
```

Comment:
But we can read subset of rows and marks by using primary key or secondary indices.

Reply:
True, the `result.parts_with_ranges is returned by ReadFromMergeTree::filterPartsByPrimaryKeyAndSkipIndexes function.  the variable part is object of RangesInDataPart. So the count of marks and rows is what we want.

Reply:
> But we can read subset of rows and marks by using primary key or secondary indices.

True, the `result.parts_with_ranges is returned by ReadFromMergeTree::filterPartsByPrimaryKeyAndSkipIndexes function. the variable part is object of RangesInDataPart. So the count of marks and rows is what we want.